File: src/main/java/net/database/AccountAPI.java
package net.database;

import client.Client;
import org.jooq.Record;
import org.jooq.Result;

import static database.jooq.Tables.ACCOUNTS;
import static database.jooq.Tables.CHARACTERS;

public class AccountAPI {

    public static Record getAccountInfo(int aid) {
        return DatabaseCore.getConnection()
                .select().from(ACCOUNTS)
                .where(ACCOUNTS.ID.eq(aid))
                .fetchOne();
    }

    public static Record getAccountInfoTemporary(int cid) {
        return getAccountInfo(DatabaseCore.getConnection()
                .select().from(CHARACTERS)
                .where(CHARACTERS.ID.eq(cid))
                .fetchOne().getValue(CHARACTERS.ACCOUNTID));
    }

    public static int getHighestLevelOnAccount(int aid) {
        Result<Record> res = DatabaseCore.getConnection()
                .select().from(CHARACTERS)
                .where(CHARACTERS.ACCOUNTID.eq(aid))
                .fetch();

        int highest = 1; // lmao
        for (Record rec : res) {
            int level = rec.getValue(CHARACTERS.LEVEL);
            if (highest < level) {
                highest = level;
            }
        }

        return highest;
    }

    public static int getCharacterCount(int aid) {
        return DatabaseCore.getConnection()
                .fetchCount(DatabaseCore.getConnection()
                        .select().from(CHARACTERS)
                        .where(CHARACTERS.ACCOUNTID.eq(aid))
                );
    }

    public static void loadNXCash(Client client) {
        client.setCash(DatabaseCore.getConnection()
                .select(ACCOUNTS.CASH).from(ACCOUNTS)
                .where(ACCOUNTS.ID.eq(client.getAccId()))
                .fetchOne().getValue(ACCOUNTS.CASH));
    }
}

File: src/main/java/net/database/AccountAPI.kt
package net.database

import client.Client
import database.jooq.Tables.ACCOUNTS
import database.jooq.Tables.CHARACTERS
import net.database.DatabaseCore.connection
import org.jooq.Record

object AccountAPI {
    fun getAccountInfo(aid: Int): Record {
        with(ACCOUNTS) {
            return connection.select().from(this)
                .where(ID.eq(aid))
                .fetchOne()
        }
    }

    fun getAccountInfoTemporary(cid: Int): Record {
        with(CHARACTERS) {
            return getAccountInfo(
                connection.select().from(this)
                    .where(ID.eq(cid))
                    .fetchOne().getValue(ACCOUNTID)
            )
        }
    }

    fun getHighestLevelOnAccount(aid: Int): Int {
        with(CHARACTERS) {
            val res = connection.select().from(this)
                .where(ACCOUNTID.eq(aid))
                .fetch()
            var highest = 1 // lmao

            for (rec in res) {
                val level = rec.getValue(LEVEL)

                if (highest < level) {
                    highest = level
                }
            }
            return highest
        }
    }

    fun getCharacterCount(aid: Int): Int {
        with(CHARACTERS) {
            return connection.fetchCount(
                connection.select().from(this).where(ACCOUNTID.eq(aid))
            )
        }
    }

    fun loadNXCash(client: Client) {
        with(ACCOUNTS) {
            client.cash = connection.select(CASH).from(this)
                .where(ID.eq(client.accId))
                .fetchOne().getValue(CASH)
        }
    }
}